File: comparative_tests/go_tests/main/basic_functions_tests.go
package main

import (
	"fmt"
	bf "go_tests/basic_functions"
)

func main() {
	const IMAGEPATH string = "../images/bread.jpg"

	// Test ReadImage
	fmt.Println("\nTest ReadImage : ")
	img := bf.ReadImage(IMAGEPATH)
	fmt.Println("Image : ", img)

	// Test GetImageDimensions
	fmt.Println("\nTest GetImageDimensions : ")
	width, height := bf.GetImageDimensions(img)
	fmt.Println("Image dimensions: ", width, "x", height)

	// Test ExtractPixels
	fmt.Println("\nTest ExtractPixels : ")
	bf.ExtractPixels(IMAGEPATH)
}

File: comparative_tests/python_tests/main/basic_functions_tests.py
from basic_functions.extract_pixels import extract_pixels
from basic_functions.get_image_dimensions import get_image_dimensions
from basic_functions.read_image import read_image


def main():
    IMAGE_PATH = "../../images/bread.jpg"

    # Test read_image
    print("\nTest read_image :")
    img = read_image(IMAGE_PATH)
    print("Image :", img)

    # Test get_image_dimensions
    if img is not None:
        print("\nTest get_image_dimensions :")
        width, height = get_image_dimensions(IMAGE_PATH)
        print("Dimensions de l'image :", width, "x", height)

    # Test extract_pixels
    print("\nTest extract_pixels :")
    extract_pixels(IMAGE_PATH)

# Run the main function
if __name__ == "__main__":
    main()
